pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -887,6 +887,10 @@ func servicesDiff(os []*model.Service, ns []*model.Service) ([]*model.Service, [
 //
 // The current algorithm to allocate IPs is deterministic across all istiods.
 func autoAllocateIPs(services []*model.Service) []*model.Service {
+	// if we are using the IP Autoallocate controller then we can short circuit this
+	if features.EnableIPAutoallocate {
```

Comment:
Hmm.. this makes sense but just thinking about what implications it may have. I guess if we think people will whole-sale migrate, this is fine. If they opt-in single SEs it may not?

Regardless, there may be a one-time transition issue going from the old to new way if we do this? since there will be some time where the status isn't allocated yet?

Reply:
Definitely something to consider wrt to transition issues. I'm not sure if "Auto IP is dropped temporarily is worse than 2 different autoIP schemas are running and there are temporary collisions as a result".

Reply:
I was thinking more about this. We can consider adding a field to the model.Service which denotes that the Auto-assigned IPs were populated from status and should be left as is rather than possibly re-assigned. It doesn't fix potential conflicts from occurring if v1 and v2 assignment are active at the same time and v2 controllers have not completed writing assignments to status for every SE

Reply:
If we do that should we just write them to the normal Address instead of the auto addresses? Not sure if it makes things easier or harder..

Reply:
So add them as regular addresses rather than auto-assigned but then denote that they are not regular addresses with a new field. I can take a look at that

Reply:
So I guess the migration path here is:
* Run new istiod for a bit with 0 clients (easy with revisions!), let it allocate all the existing SEs
* Now when proxies connect they will get the correct result

Even without revisions, in _most_ cases this will work since it is quick to update all the SEs, and it takes time for clients to start connecting.

does that sound right to you?

Reply:
This does sound like a good strategy and tbh, even with the old code (as it exists minus this PR) that is probably a much better idea than just turn it on and have the potential for collisions between v1 and v2 while the controller is churning through all the SE assigning/writing status